### PR TITLE
feat: add help command

### LINKED ISSUE #29: feat: help command
We should add a help command which explains the process, how to use the bot, etc. Additionally, this should take an optional arg that offers help for a single command.

### PR DESCRIPTION
**Pull request by nullishamy**
Closes #29

### CODE REVIEW

File: src/commands/help.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { ProjectsClient } from '../client'
+import config from '../config'
+import { Command } from '../managers/commands'
+import { embeds } from '../utils/embeds'
+
+const help: Command = {
+  name: 'help',
+  description: 'Displays help for the bot, or a given command',
+  permissionLevel: 'veterans',
+  guildId: '985644694736359544',
```
Comment (BobobUnicorn) [inline src/commands/help.ts:11]: why is this hard-coded

File: src/config.ts [hunk 2]
```diff
@@ -16,16 +28,62 @@ const config = {
...
-      { name: 'Invalid link (Not GitHub or GitLab)', value: 'invalid-repository' },
-      { name: 'Invalid user ID', value: 'invalid-id' },
-      { name: 'Plagiarism', value: 'plagiarism' },
-      { name: 'Advertisement', value: 'ad' }
-    ] as const
+    descriptions
+  },
+  help: {
+    commands: {
+      edit: '',
```
Comment (BobobUnicorn) [inline src/config.ts:35]: add a note about adding help text if required

File: src/config.ts [hunk 3]
```diff
@@ -16,16 +28,62 @@ const config = {
...
+      reject: '',
+      help: 'When ran without a command, the help is provided as an ephemeral message.',
+      create: ''
+    },
+    general:
+`
+**General**
+  ⮞ Applying the final vote on a project (3up / 3down) will action that and remove it from <#813496732901965924>
```
Comment (BobobUnicorn) [inline src/config.ts:43]: 3up/3down depends on the config, no?

File: src/config.ts [hunk 4]
```diff
@@ -16,16 +28,62 @@ const config = {
...
+**Submitting your own project**
+You may submit your own project under the following terms:
+  ⮞ You do not vote on your own project
+  ⮞ You do not pre-emptively read feedback and make changes
+  ⮞ You do not interfere with feedback
+
+
+**Miscellaneous**
+  ⮞ User mention format: \`<@ID>\`
+  ⮞ You can assign viewer roles through <#987102838171770972>
```
Comment (BobobUnicorn) [inline src/config.ts:78]:
```suggestion
  ⮞ You can assign reviewer roles through <#987102838171770972>
```

File: src/utils/embeds.ts [hunk 5]
```diff
@@ -49,7 +51,40 @@ const privateSubmission = async (project: IncomingSubmissionData, author: User):
...
+    title: `Help for command ${command.name}`,
+    description,
+    timestamp: new Date(),
+    color: 0xFFFFFF
+  })
+}
+
+const generalHelp = (): Discord.MessageEmbed => {
+  return new Discord.MessageEmbed({
+    title: 'How to use Projects Bot',
```
Comment (BobobUnicorn) [inline src/utils/embeds.ts:78]:
```suggestion
    title: 'How to use Projects Showcase',
```

File: src/utils/embeds.ts [hunk 6]
```diff
@@ -49,7 +51,40 @@ const privateSubmission = async (project: IncomingSubmissionData, author: User):
...
+    description: config.help.general,
+    timestamp: new Date(),
+    color: 0xFFFFFF
```
Comment (BobobUnicorn) [inline src/utils/embeds.ts:81]: 7289DA

**Review by BobobUnicorn — CHANGES_REQUESTED** [review]

File: src/commands/help.ts (hunk 1 above)
Comment (nullishamy) [reply src/commands/help.ts:11]: As part of the commands implementation, IDs were hard coded until a better solution was implemented. This has been resolved in feat/beta-preparation.

**Review by BobobUnicorn — APPROVED** [review]

**Pull request merged**